lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -273,7 +260,10 @@ public function diffTable(Table $table1, Table $table2)
                 if ($this->diffForeignKey($constraint1, $constraint2) === false) {
                     unset($fromFkeys[$key1], $toFkeys[$key2]);
                 } else {
-                    if (strtolower($constraint1->getName()) === strtolower($constraint2->getName())) {
+                    $name1 = $constraint1->getName();
+                    $name2 = $constraint2->getName();
+
+                    if ($name1 !== null && $name2 !== null && strcasecmp($name1, $name2) === 0) {
```

Comment:
This looks incorrect. Should be something like: https://github.com/doctrine/dbal/blob/34c3ad11496d8065d164fc095ac5569ebbc3d74f/lib/Doctrine/DBAL/Schema/Comparator.php#L436